src/domain/wallets/tx-history.ts

```diff
@@ -65,92 +68,143 @@ export const fromLedger = (
+      if (type == LedgerTransactionType.IntraLedger && paymentHash) {
+        txType = ExtendedLedgerTransactionType.LnIntraLedger
+      }
+
+      let walletTransaction: WalletTransaction
+      switch (txType) {
+        case ExtendedLedgerTransactionType.IntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username as Username,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username as Username,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.OnchainIntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.OnChain,
+              address,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username || null,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.OnchainPayment:
+        case ExtendedLedgerTransactionType.OnchainReceipt:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.OnChain,
+              address,
+            },
+            settlementVia: {
+              type: SettlementMethod.OnChain,
+              transactionHash: txHash as OnChainTxHash,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.LnIntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.Lightning,
+              paymentHash: paymentHash as PaymentHash,
+              pubkey: pubkey as Pubkey,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username || null,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.Payment:
+        case ExtendedLedgerTransactionType.Invoice:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.Lightning,
+              paymentHash: paymentHash as PaymentHash,
+              pubkey: pubkey as Pubkey,
+            },
+            settlementVia: {
+              type: SettlementMethod.Lightning,
+              paymentSecret: null,
+            },
+          }
+          return walletTransaction
+      }
+
+      walletTransaction = {
```

Comment:
shouldn't this also be part of the `case:` scenario?

Reply:
that would be cleaner that assuming a default?

Reply:
This part of the code was a little messy and needs to be cleaned up a bit. It's tricky because we have send/receive types mixed in with things like `"routing_fee"` and `"exchange_rebalance"` under the same `LedgerTransactionType` definition. I tried a few things but ended up in some dead-ends, so opted to maintain the current behaviour which defaults to Intraledger initiated and settled and then to clean this up in another later PR (to not block the rest of Samer's integration work).

Reply:
no strong opinion on this except that `"routing_fee"` and `"exchange_rebalance"` are not meant to be returned to any normal user.

it's a transaction that is related to the admin/"bank owner", so this won't be expose to the public API in my mind.

Reply:
Yup exactly, we should probably differentiate those into another domain type